biz.aQute.bndlib/src/aQute/bnd/build/Workspace.java

```diff
@@ -664,6 +709,30 @@ public void syncCache() throws Exception {
 		return plugins;
 	}
 
+	/**
+	 * Get the repositories and ensure they are all ready.
+	 *
+	 * @return a promise with the list of repos
+	 */
+	public Promise<List<RepositoryPlugin>> getInitializedRepositories() {
+		try {
+			List<RepositoryPlugin> repositories = getRepositories();
+			List<Promise<Void>> promises = new ArrayList<>();
+			for (RepositoryPlugin repo : repositories) {
+				promises.add(repo.sync());
+			}
+			return Promises.all(promises)
+				.map(l -> {
+					return repositories;
+				});
+
+			// failures should be visible on the repositories,
+			// this is just about syncing
+		} catch (Exception e) {
+			return Promises.failed(e);
```

Comment:
You should not be using `Promises`. Use `getPromiseFactory().failed()` instead. This properly uses the executors configured in the promise factory.